### PR TITLE
Project Labyrinth - Joel Harde and Rebeka Kovacic

### PR DESCRIPTION
**Pull request by JHarde**
Link to my deployed project: https://rejoel-labyrinth.netlify.app/

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/components/Input.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// --- STYLED COMPONENTS ---
+
+const FormContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`;
+
+const FormContent = styled.form``;
+
+const FormLabel = styled.label``;
```
Comment (luuf81) [inline code/src/components/Input.js:17]: this is not doing any styling and can be removed it seems.

File: code/src/reducers/fetch.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      .then((data) => {
+        dispatch(game.actions.setGame(data));
+
+        dispatch(ui.actions.setLoading(false));
+      });
+  };
+};
+
+//function that returns another function aka THUNKy
+//why not curlies around the direction?
```
Comment (luuf81) [inline code/src/reducers/fetch.js:24]: It's just a regular parameter right? You are not decomposing any object here and therefore no need for curlys we believe.

**Review by luuf81 — APPROVED** [review]
Really great job on the labyrinth project Joel & Rebeka!

- We were impressed by your use of redux and thunks. Also that you implemented the loading indicator. 
- Great use of styled components & reusable components
- Overall very clear & structured code, with lots of comments to explain 👍
- You made the site responsive
- You implemented go-back/history which for us was a bit tricky :)
- Great use of combining react & redux in your site

Overall we have no hesitation as to mark your project as approved :)

Well done! 🙌
//Marjaana & Olof

File: code/src/components/LoadingIndicator.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { Loading } from "react-loading-dot";
+
+export const LoadingIndicator = () => {
+  const loading = useSelector((state) => state.ui.loading);
+  return <>{loading && <Loading background="#A69B8D" />}</>;
```
Comment (EmOlsz) [inline code/src/components/LoadingIndicator.js:7]: No need to use React.Fragment here

File: code/src/reducers/fetch.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import { game } from "./game";
+import { ui } from "./ui";
+
+export const fetchGame = () => {
+  return (dispatch, getStore) => {
+    dispatch(ui.actions.setLoading(true));
+
+    fetch("https://wk16-backend.herokuapp.com/start", {
+      method: "POST",
+      body: JSON.stringify({ username: getStore().game.username }),
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/reducers/fetch.js:11]: @luuf81 This is redux thunk, it always takes two arguments; dispatch as required argument and store as optional argument

**Comment by JHarde** [comment]
We have fixed most of the requested changes now! Have a few questions though that you perhaps can clarify for us :)

1. When we add ./ to our path in Actions.js line 9 it doesn't respond to that, that's why we kept it.
2. We assumed that the root element in Actions.js <DirectionWrap> could only have one key so we added key={action.direction}. We had another key which was {action.desription} that we didn't know where to move since <DirectionWrap> is the root element. We deleted this.
3. In LoadingIndicator when we remove the React fragments the code is not executing. We tried to wrap our return in parenthesis but that didn't help.

Perhaps you can guide us a bit or just clarify why these problems occur.

**Comment by EmOlsz** [comment]
Hello @JHarde! Thank you for your descriptive message, let me reference to your questions :

1. You are right, it's my bad - when to do not add `./` in path to this component, JS will know that it should look for `lib` folder in project root (so, at the highest level). So, no other actions are required from your side.
2. Good, you provided correct fix - we needed to delete `React.Fragment`, add `key` to `<DirectionWrap />` and remove `key` from `<DirectionButton />`. So, no other actions are required from your side.
3. Hmm, I think we have only two correct options for this file :

```
First one :

return <>{loading && <Loading background="#A69B8D" />}</>;
```

or 

```
Second one :

return loading && <Loading background="#A69B8D" />;
```


Let me know how does number 3. works for you, apart from that all of required changes are done correctly!

**Comment by JHarde** [comment]
Pushed the last change now! In LoadingIndicator it seemed as if both the first and second option worked, changed it to the second one so got rid of the fragments :)

**Comment by EmOlsz** [comment]
Super, thank you, project approved :)

**Pull request closed without merging**